src/ui/components/Queue.tsx

```diff
@@ -1,3 +1,6 @@
+// REVIEW: I feel like this file has become quite chockfull of responsabilities.
```

Comment:
Very much agree.

I'd love to see this file modulized and properly unit tested, considering the wide responsibilities it has

Reply:
I think it's reasonable to tackle this on a follow-up PR.

Reply:
definitely! let's not address this here!